ui/src/services/booking-service.ts

```diff
@@ -0,0 +1,39 @@
+import { makeAutoObservable } from 'mobx'
+
+import { deviceBySerialStore } from '@/store/device-by-serial-store'
+
+import { groupService } from './group-service'
+
+export class BookingService {
+  bookedBeforeTime: string | undefined
+
+  constructor(serial: string) {
+    makeAutoObservable(this)
+
+    this.init(serial)
+  }
+
+  async init(serial: string): Promise<void> {
+    const { data: device } = deviceBySerialStore.deviceQueryResult(serial)
+
+    if (device?.statusChangedAt && device.bookedBefore) {
+      this.setTime(device.statusChangedAt, device.bookedBefore)
+    }
+  }
+
+  async reBookDevice(serial: string): Promise<void> {
+    const device = await deviceBySerialStore.fetch(serial)
+
+    groupService.invite(device).then(() => {
```

Comment:
I'm leaning towards avoiding .then syntax in the new frontend for consistency, but I'm open to hearing other perspectives. What do you think?

Reply:
I’ve been leaning towards using `async/await` for consistency as well, but using it forces the function to be "colored" red (as asynchronous), which can spread that async nature to other functions that call it. This is discussed in more detail in the article ["What Color is Your Function?"](https://journal.stuffwithstuff.com/2015/02/01/what-color-is-your-function/). If you're looking to avoid that, `.then` might be a cleaner option in some cases.